registry/client/transport/http_reader.go

```diff
@@ -0,0 +1,167 @@
+package transport
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/docker/distribution"
+)
+
+// NewHTTPReadSeeker handles reading from an HTTP endpoint using a GET
+// request. When seeking and starting a read from a non-zero offset
+// the a "Range" header will be added which sets the offset.
+func NewHTTPReadSeeker(client *http.Client, url string, size int64) distribution.ReadSeekCloser {
```

Comment:
This package may not be the ideal place. We don't have a solid place for this yet, do we can disregard. Just a TODO here will absolve guilt.